### PR TITLE
remove uuid from SignInData

### PR DESCRIPTION
**Pull request by OrenZak**
Separated Kin.start, one for WhitelistData and one for jwt String:
1. `Kin.start(@NonNull Context appContext, @NonNull WhitelistData whitelistData)`
2. `Kin.start(@NonNull Context appContext, @NonNull String jwt)`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,32 +100,27 @@ As can be seen in the sample app, there are just few step required to integrate
        }
     
        ```
-1. Create SignInData object and set userID, appID, apiKey etc.
+1. Create WhitelistData object and set userID, appID, apiKey etc or just pass a JWT String on `Kin.start`.
```
Comment (YohayBar) [inline README.md]: or just pass a registration JWT String on `Kin.start`

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/Kin.java [hunk 2]
```diff
@@ -47,8 +50,40 @@ private static Kin getInstance() {
...
+            init(appContext, signInData);
+        }
+    }
+
+    private static SignInData getWhiteListSignInData(@NonNull final WhitelistData whitelistData) {
+        SignInData signInData = new SignInData()
+            .signInType(SignInTypeEnum.WHITELIST)
+            .userId(whitelistData.getUserID())
+            .appId(whitelistData.getAppID())
+            .apiKey(whitelistData.getApiKey());
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/Kin.java]: we still need UUID for SignInData we just need to generated it ourselves

File: app/src/main/java/com/ecosystem/kin/app/App.java [hunk 3]
```diff
@@ -19,25 +19,30 @@ public void onCreate() {
 
         Fabric.with(this, new Crashlytics());
 
-        SignInData signInData;
-
         if (BuildConfig.IS_JWT_REGISTRATION) {
             /*
             * SignInData should be created with registration JWT {see https://jwt.io/} created securely by server side
-            * In the the this example 'SignInRepo.getJWTSignInData' receive an empty/null JWT String and the sample app generate the JWT locally.
+            * In the the this example 'SignInRepo.getJWT' receive an empty/null JWT String and the sample app generate the JWT locally.
```
Comment (YohayBar) [inline app/src/main/java/com/ecosystem/kin/app/App.java]: update "receive an empty/null JWT String and the sample app generate"

File: app/src/main/java/com/ecosystem/kin/app/App.java [hunk 4]
```diff
@@ -19,25 +19,30 @@ public void onCreate() {
...
+            String jwt = SignInRepo.getJWT(this);
+
+            try {
+                Kin.start(getApplicationContext(), jwt);
+            } catch (InitializeException e) {
+                e.printStackTrace();
+            }
         } else {
             //Use whitelist signing data for small scale testing
-            signInData = SignInRepo.getWhitelistSignInData(this, getAppId(), getApiKey());
```
Comment (YohayBar) [inline app/src/main/java/com/ecosystem/kin/app/App.java]: update //Use whitelist signing data for small scale testing

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/Kin.java [hunk 5]
```diff
@@ -83,6 +118,7 @@ private static void registerAccount(@NonNull final Context context, @NonNull fin
         try {
             publicAddress = getPublicAddress();
             signInData.setWalletAddress(publicAddress);
+            signInData.setDeviceId(UUID.randomUUID().toString());
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/Kin.java]: do we need to reuse UUID.randomUUID() i.e save it in cache?

File: kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/model/WhitelistData.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+
+        WhitelistData that = (WhitelistData) o;
+
+        if (userID != null ? !userID.equals(that.userID) : that.userID != null) {
```
Comment (YohayBar) [inline kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/model/WhitelistData.java]: if userID != null && userID.equals(that.userID) 

or Objects.equals()?

Comment (OrenZak) [reply kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/model/WhitelistData.java]: we can't use Objects.equals()

Comment (YohayBar) [reply kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/model/WhitelistData.java]: why not using && if userID!=null checking !userID.equals(that.userID) is enough

Comment (OrenZak) [reply kin-ecosystem-sdk/src/main/java/com/kin/ecosystem/data/model/WhitelistData.java]: it's generated from IntelliJ and I didn't change that, is that so important?

**Pull request merged**